cfgov/login/forms.py

```diff
@@ -78,94 +22,29 @@ class LoginForm(AuthenticationForm):
     }
 
-    def clean(self):
-        try:
-            return super().clean()
-        except (LockoutError, PasswordExpiredError) as e:
-            # The user authenticated correctly, but either they are under a
-            # temporary lockout or their password has expired.
-            # Propogate the exception without recording this as a failed login.
-            raise e
-        except ValidationError as e:
-            # The Django AuthenticationForm raised a validation error because
-            # it could not authenticate the username/password combination.
-            #
-            # We want to record the failed login attempt, and maybe raise a
-            # lockout validation error of our own before raising Django's
-            self.record_failed_login_attempt(self.cleaned_data["username"])
-            raise e
-
-    def record_failed_login_attempt(self, username):
-        UserModel = get_user_model()
-
-        try:
-            user = UserModel._default_manager.get(username=username)
-        except ObjectDoesNotExist:
-            raise self.get_invalid_login_error() from None
-
-        fa, created = FailedLoginAttempt.objects.get_or_create(user=user)
-        now = time.time()
-        fa.failed(now)
-
-        # Defaults to a 2 hour lockout for a user
-        time_period = now - settings.LOGIN_FAIL_TIME_PERIOD
-        attempts_allowed = settings.LOGIN_FAILS_ALLOWED
-
-        if fa.too_many_attempts(attempts_allowed, time_period):
-            dt_now = timezone.now()
-            lockout_expires = dt_now + timedelta(
-                seconds=settings.LOGIN_FAIL_TIME_PERIOD
-            )
-            lockout = user.temporarylockout_set.create(
-                expires_at=lockout_expires
-            )
-            lockout.save()
-            raise LockoutError(
-                self.error_messages["temporary_lock"],
-                code="temporary_lock",
-            )
-        else:
-            fa.save()
-
     def confirm_login_allowed(self, user):
         super().confirm_login_allowed(user)
 
-        # Check to make sure the user is not already locked out.
-        self.check_for_lockout(user)
-
         # Check to make sure the user's password hasn't expired.
         self.check_for_password_expiration(user)
 
-        # Finally, clear any failed login attempts
-        self.clear_failed_login_attempts(user)
-
-    def check_for_lockout(self, user):
-        now = timezone.now()
-        lockout_query = user.temporarylockout_set.filter(expires_at__gt=now)
-        if lockout_query.count() > 0:
-            raise LockoutError(
-                self.error_messages["temporary_lock"],
-                code="temporary_lock",
-            )
-
     def check_for_password_expiration(self, user):
-        dt_now = timezone.now()
-        try:
-            current_password_data = user.passwordhistoryitem_set.latest()
+        # Superuser passwords never expire.
+        if user.is_superuser:
+            return
 
```

Comment:
```suggestion
```

I'm having second thoughts about introducing this in my PR. I think I misinterpreted [this logic](https://github.com/cfpb/consumerfinance.gov/blob/7b6190d4c43c8fcffb5a126e1285ecb1e731344d/cfgov/login/signals.py#L28-L30) in the existing code: the current logic doesn't prevent superuser passwords from expiring. If anything we should be more concerned that they expire!